src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java

```diff
@@ -95,6 +98,20 @@ public void set(BinlogPosition p) {
 		position.set(p);
 	}
 
+	public void setSync(BinlogPosition p) {
+		LOGGER.debug("syncing binlog position: " + p);
+		position.set(p);
+		while ( true ) {
+			thread.interrupt();
+			BinlogPosition s = storedPosition.get();
+			if ( p.newerThan(s) ) {
```

Comment:
the newerThan makes sense at other places, but confused me here a bit. It's essentially (p != s) here.